# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   ### The z-test should be appropriate here, as the sample size is high enough for the Central Limit Theorem would apply.
       
   2. What are the null and alternate hypotheses?
   ### The null hypothesis is that the mean rate of call-backs is equal for black resumes and white resumes. The alternate hypothesis is that the rate is different for resumes of different races.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
%matplotlib inline
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
from matplotlib import mlab

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [5]:
data.columns

Index(['id', 'ad', 'education', 'ofjobs', 'yearsexp', 'honors', 'volunteer',
       'military', 'empholes', 'occupspecific', 'occupbroad', 'workinschool',
       'email', 'computerskills', 'specialskills', 'firstname', 'sex', 'race',
       'h', 'l', 'call', 'city', 'kind', 'adid', 'fracblack', 'fracwhite',
       'lmedhhinc', 'fracdropout', 'fraccolp', 'linc', 'col', 'expminreq',
       'schoolreq', 'eoe', 'parent_sales', 'parent_emp', 'branch_sales',
       'branch_emp', 'fed', 'fracblack_empzip', 'fracwhite_empzip',
       'lmedhhinc_empzip', 'fracdropout_empzip', 'fraccolp_empzip',
       'linc_empzip', 'manager', 'supervisor', 'secretary', 'offsupport',
       'salesrep', 'retailsales', 'req', 'expreq', 'comreq', 'educreq',
       'compreq', 'orgreq', 'manuf', 'transcom', 'bankreal', 'trade',
       'busservice', 'othservice', 'missind', 'ownership'],
      dtype='object')

3\. Compute margin of error, confidence interval, and p-value.

In [6]:
print(len(data.race == 'b'))
print(len(data.race == 'w'))

4870
4870


In [7]:
from scipy.stats import ttest_ind

blackcalls = data.call[data.race == 'b']
whitecalls = data.call[data.race == 'w']

tstat, p = ttest_ind(blackcalls, whitecalls)

print('p =', p)

p = 3.94080210313e-05


In [8]:
stderrorblk = np.sqrt(np.mean(blackcalls)*(1-np.mean(blackcalls))/4870)
stderrorwht = np.sqrt(np.mean(whitecalls)*(1-np.mean(whitecalls))/4870)
margin = np.sqrt(stderrorblk**2 + stderrorwht**2)*2 #95 percent confidence interval
                                                    #gives alpha approx. 2.0
confidence_interval = (np.mean(data.call) - margin, np.mean(data.call) + margin)
print(margin)
print(confidence_interval)
print(np.mean(data.call))

0.0110073482438
(0.069485468621657456, 0.09150016510923091)
0.08049281686544418


### The margin of error is 1.1%. Therefore we can state with 95% confidence that the difference between the population callback rate is at least 7%. With a callback ratio of only 8%, this is a significant difference!

5\. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

### We cannot yet say if racial-sounding names are the most important factor in callback success. We only have seen that race is apparently one factor. To determine the most important factor, we would look at the correlation coefficient with callback rate for each feature.